zebra-chain/src/primitives/redpallas/scalar_mul.rs

```diff
@@ -0,0 +1,207 @@
+// -*- mode: rust; -*-
+//
+// This file is part of redpallas.
+// Copyright (c) 2019-2021 Zcash Foundation
+// Copyright (c) 2017-2021 isis agora lovecruft, Henry de Valence
+// See LICENSE for licensing information.
+//
+// Authors:
+// - isis agora lovecruft <isis@patternsinthevoid.net>
+// - Henry de Valence <hdevalence@hdevalence.ca>
+// - Deirdre Connolly <deirdre@zfnd.org>
+
+use std::{borrow::Borrow, fmt::Debug};
+
+use group::Group;
+use halo2::{arithmetic::FieldExt, pasta::pallas};
+
+pub trait NonAdjacentForm {
+    fn non_adjacent_form(&self, w: usize) -> [i8; 256];
+}
+
+/// A trait for variable-time multiscalar multiplication without precomputation.
+pub trait VartimeMultiscalarMul {
+    /// The type of point being multiplied, e.g., `AffinePoint`.
+    type Point;
+
+    /// Given an iterator of public scalars and an iterator of
+    /// `Option`s of points, compute either `Some(Q)`, where
+    /// $$
+    /// Q = c\_1 P\_1 + \cdots + c\_n P\_n,
```

Comment:
A silly question: why those '\\' in the expression? I'm used to Latex with lowering indices without backslashes.

Reply:
I think to make the LaTeX work well with Rustdoc and KaTeX to render it on the fly in the browser